nixos/modules/config/system-environment.nix

```diff
@@ -8,6 +8,28 @@ let
 
   cfg = config.environment;
 
+  pamVariables =
+    let
+      absoluteVariables = mapAttrs (const toList) cfg.sessionVariables;
+
+      pamProfiles =
+        map
+          (replaceStrings ["$HOME" "$USER"] ["@{HOME}" "@{PAM_USER}"])
```

Comment:
I'm not so fond of this but couldn't think of a better solution.

Perhaps it would be better to have a PAM specific `profileRelativeEnvVars`. Then one could have better control over which variables end up in the PAM configuration as well. For systemd to find the service-files it is sufficient to have `XDG_DATA_DIRS`.

Reply:
I ended up adding `profileRelativeSessionVariables` which is PAM specific, but I still have the above `replaceStrings` since I wanted to avoid duplicating `environment.profiles`.

Reply:
I think this is good.